### PR TITLE
Use devise-i18n for translating devise views

### PR DESCRIPTION
**Pull request by jonatanklosko**
Recently, Christine has wanted to apply some changes to the Czech translation and I found out that they were related to the `devise` gem. Naturally I thought it's the [devise-i18n](https://github.com/tigrish/devise-i18n) what we need to update and that's what we did. The new version has been released and I was about to update the version used on our side when I realised that we don't actually use the library directly. Confusingly enough, phrases really seemed to come from this gem. Looking at other `devise` related gems we use, I stumbled upon [devise-bootstrap-views](https://github.com/hisea/devise-bootstrap-views), and [as you can see](https://github.com/hisea/devise-bootstrap-views/tree/master/locales) it includes it's own set of locale files that come from `devise-i18n`, except they are not updated over time...

TL;DR

To have up-to-date translations from `devise-i18n` and keep using `devise-bootstrap-views` without including a bunch of files in our repo, I added the former as a dependency and intentionally placed it after `devise-bootstrap-views`. Given that Bundler loads gems as they appear in the Gemfile, this order makes `devise-i18n` phrases take precedence.

### CODE REVIEW

**Comment by jfly** [comment]
This seems like a great idea! There are a surprisingly large number of travis errors, maybe related?

**Comment by jonatanklosko** [comment]
Yeah, these are related to wording change in `devise-i18n`. I think it was for no good reason, so I created an issue over here: https://github.com/tigrish/devise-i18n/issues/228

**Comment by jonatanklosko** [comment]
I had to override the "Log in" phrase anyway as it was impossible upstream, see [this comment](https://github.com/tigrish/devise-i18n/issues/228#issuecomment-370566969). Maybe you have a batter idea of doing this?

**Comment by jfly** [comment]
That sounds good to me. I'm also ok with changing our wording to "log in" to stay consistent with devise upstream, but I guess that would involve updating a bunch of our tests?

**Comment by jonatanklosko** [comment]
Well, I'm still confused why they decided to use "Log in" there. For instance, have a look at [this file](https://github.com/plataformatec/devise/blob/master/config/locales/en.yml), you can find 4 occurrences of `/.*sign in.*/` there, but no `/.*log in.*/` at all.
Whatever you're ok with!

**Comment by jfly** [comment]
Weird. I like what you've got right now =)
